SignalServiceKit/src/Messages/Interactions/TSOutgoingMessage.m

```diff
@@ -553,6 +563,11 @@ - (void)updateWithDeliveredRecipient:(NSString *)recipientId
     OWSAssert(recipientId.length > 0);
     OWSAssert(transaction);
 
+    // If delivery notification doesn't include timestamp, use "now" as an estimate.
```

Comment:
When it's a client generated delivery receipt (which we aren't using, the client should specify), when it's a server generated delivery receipt, we use the timestamp on the envelope.

I can't think of a situation where this would happen and not be a bug, can we fail and log?

Reply:
Sorry - I was confusing read receipts with delivery receipts. I believe this is correct as written.